ui/service.go

```diff
@@ -281,9 +280,10 @@ type Service interface {
 	ListUserGroupChannels(token, groupID string, page, limit uint64) (b []byte, err error)
 
 	// Publish facilitates a thing publishin messages to a channel.
-	Publish(token, thKey string, msg *messaging.Message) error
-	// ReadMessage facilitates a thing reading messages published in a channel.
-	ReadMessage(token, chID, thKey string, page, limit uint64) ([]byte, error)
+	Publish(token, chID, thKey, message string) error
+	// ReadMessages retrieves messages published in a channel.
+	ReadMessages(token, chID, thKey string, page, limit uint64) ([]byte, error)
```

Comment:
```suggestion
	ReadMessages(token, chID string, page, limit uint64) ([]byte, error)
```
No need for thingKey

Reply:
the thing key is passed back to the HTML template to be used by the t`est channel modal` for publishing to the channel